app/scripts/controllers/persistentVolumeClaim.js

```diff
@@ -30,14 +30,25 @@ angular.module('openshiftConsole')
       }
     ];
 
+    $scope.storageClassesVersion = APIService.getPreferredVersion('storageclasses');
     $scope.pvcVersion = APIService.getPreferredVersion('persistentvolumeclaims');
     $scope.eventsVersion = APIService.getPreferredVersion('events');
+    $scope.isExpansionAllowed = false;
 
+    var storageClass = $filter('storageClass');
     var watches = [];
 
     var pvcResolved = function(pvc, action) {
       $scope.pvc = pvc;
       $scope.loaded = true;
+      var storageClassName = storageClass($scope.pvc);
+
+      if(storageClassName) {
```

Comment:
Only request the storage class again if the value has changed.

Make sure we handle the case when there's no storage class. Right now it looks like isExpansionAllowed is never set when there is no storage class on the PVC.

```jsx
var storageClass;
var updateStorageClass = function(pvc) {
  var storageClassName = storageClass($scope.pvc);
  if (storageClassName !== _.get(storageClass, 'metadata.name')) {
    DataService.get($scope.storageClassesVersion, storageClassName, {}).then(function(sc) {
      storageClass = sc;
    });
  }
};

var pvcResolved = function(pvc, action) {
  $scope.pvc = pvc;
  $scope.loaded = true;
  updateStorageClass(pvc);
  $scope.isExpansionAllowed = (!storageClass || storageClass.allowVolumeExpansion) && pvc.status.phase === 'Bound';
};
```